Neo4j.Driver/Neo4j.Driver/Public/Types/Vector.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) "Neo4j"
+// Neo4j Sweden AB [https://neo4j.com]
+// 
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Neo4j.Driver.Internal;
+using Neo4j.Driver.Internal.Types;
+
+namespace Neo4j.Driver;
+
+/// <summary>
+/// An abstract base class for a mathematical vector with elements of supported numeric types.
+/// </summary>
+/// <remarks>
+/// Supported element types are: <see cref="float"/>, <see cref="double"/>, <see cref="sbyte"/>, <see cref="short"/>,
+/// <see cref="int"/>, and <see cref="long"/>.
+/// </remarks>
+public abstract class Vector : IValue, IVector, IEquatable<IVector>
+{
+    public static readonly HashSet<Type> SupportedTypes =
+    [
+        typeof(float), // f32
+        typeof(double), // f64
+        typeof(sbyte), // i8
+        typeof(short), // i16
+        typeof(int), // i32
+        typeof(long) // i64
+    ];
+
+    /// <summary>
+    /// Determines whether the specified type is supported for use as a vector element.
+    /// </summary>
+    /// <param name="type">The type to check for support.</param>
+    /// <returns>
+    /// <c>true</c> if the specified type is supported; otherwise, <c>false</c>.
+    /// </returns>
+    public static bool IsSupported(Type type)
+    {
+        return SupportedTypes.Contains(type);
+    }
+
+    /// <summary>
+    /// Gets the elements of the vector as an array of objects, regardless of their underlying type.
+    /// </summary>
+    public abstract IEnumerable<object> UntypedValues { get; }
+
+    /// <summary>
+    /// Gets the original byte stream from which the vector was deserialized, if applicable.
+    /// </summary>
+    public abstract byte[] OriginalByteStream { get; }
```

Comment:
Perhaps we should convert from the UntypedValues here rather than store the bytestream

Reply:
I wouldn't want to reverse-engineer the byte stream from the untyped values, I don't think.